README.md

```diff
@@ -1,40 +1,125 @@
-# SPDX-Python-LicenseMatching
-Implement SPDX License Matching in Python. Project in CommunityBridge Linux Foundation 2020. 
-The Software Package Data Exchange (SPDX) specification is a standard format for communicating the components, licenses and copyrights associated with a software package. This Project is mainly based on implementing License Matching Techniques in Python. 
+# SPDX Python Library for LicenseMatching
+A Python Library to implement SPDX License Matching. This Project is a result of CommunityBridge Linux Foundation 2020 contribution by [@anshuldutt21](https://github.com/anshuldutt21/).
 
-## Check if 2 License Texts match
-License Texts are matched based on the [SPDX License Matching Guidelines](https://spdx.dev/license-list/matching-guidelines/). In order to implement License Matching between 2 License Texts, do the following steps.
+Home: https://github.com/anshuldutt21/spdx_python_licensematching/
+
+Issues: https://github.com/anshuldutt21/spdx_python_licensematching/issues
+
+# License
+[Apache-2.0](https://github.com/spdx/tools-python/blob/master/LICENSE)
+
+# Features
+
+*    An Interface which compares text against a license template using the license matching guidelines.
+*    An Interface which returns all matching SPDX listed license ID's for any license text.
+*    An interface which compares 2 license texts and returns a boolean indicating if the 2 licenses match per the license matching guidelines.
+*    When there is no match, a return value is provided to describe where and why the license does not match.
 
-Clone the Repository and move to the spdx-python-licensematching folder.
+# Installation
 
-Move the files in the normalize_license_text folder.
+As always you should work in a virtualenv or venv. Do the following steps to setup the Python Package in your system.
 
-Execute the command 
 ```bash
-Python normalize.py <File1> <File2>
+git clone https://github.com/anshuldutt21/spdx_python_licensematching.git
+cd spdx_python_licensematching
+pip install -e .
 ```
-## Compare a License and a Template
-License templates are implemented on the basis of the [SPDX License Matching Guidelines](https://spdx.dev/license-list/matching-guidelines/). In order to use the feature  of comparing a license text and a template-
 
-```bash
-cd compare_template_text/
+# Examples
+
+There are some examples for how to use the code in the examples folder in the directory. Go to the Folder and run the following files to see the output.
+
+```python
+python compare_license_texts.py
+```
+
+This file will compare any 2 sample texts which can be passed into the module and return the output.
+
+```python
+python compare_license_template_text.py
+```
+This file parses the sample text and the template provided in the example and matches them producing the desired output.
+
+```python
+python compare_all_ids.py
+```
+This file parses the sample text and matches it against all the license ids and returns matched licenses.
+
```

Comment:
While running the examples file on my system I got

> python3 compare_license_texts.py
There was some problem with the function.
Traceback (most recent call last):
  File "compare_license_texts.py", line 15, in <module>
    stringtemplate = object_a.returnfinalstring()
  File "/Users/rlodha/spdx-python-licensematching/normalize_license_text/normalize_class.py", line 30, in returnfinalstring
    normalized_string = self.copyrightsymbol(normalized_string)
  File "/Users/rlodha/spdx-python-licensematching/normalize_license_text/normalize_class.py", line 89, in copyrightsymbol
    normalized_string = normalized_string.replace("copyright", '')
AttributeError: 'NoneType' object has no attribute 'replace'

Reply:
> python3 compare_license_template_text.py
There was some problem with the function.
Traceback (most recent call last):
  File "compare_license_template_text.py", line 15, in <module>
    stringtemplate = object_a.returnfinalstring_for_template()
  File "/Users/rlodha/spdx-python-licensematching/normalize_license_text/normalize_class.py", line 48, in returnfinalstring_for_template
    normalized_string_for_template)
  File "/Users/rlodha/spdx-python-licensematching/normalize_license_text/normalize_class.py", line 128, in punctuation
    normalized_string = normalized_string.replace('_', '-')
AttributeError: 'NoneType' object has no attribute 'replace'

> python3 compare_all_ids.py
There was some problem with the function.
Traceback (most recent call last):
  File "compare_all_ids.py", line 22, in <module>
    normalized_text_string = object_a.returnfinalstring_for_template()
  File "/Users/rlodha/spdx-python-licensematching/normalize_license_text/normalize_class.py", line 48, in returnfinalstring_for_template
    normalized_string_for_template)
  File "/Users/rlodha/spdx-python-licensematching/normalize_license_text/normalize_class.py", line 128, in punctuation
    normalized_string = normalized_string.replace('_', '-')
AttributeError: 'NoneType' object has no attribute 'replace'

Reply:
@rtgdk It's working if we just use python compare_license_texts.py rather than python3. I don't know the exact reason why this is happening but it might be because while setting up it uses a default python version in the environment and that's why it's not taking into account python3. Could you please tell me if it works after doing these steps. 
Did you setup the project by using pip install -e .
And Could you also try using python3.7 as that's the python version I am using.

Reply:
Using `python` uses python2 on my system. I created a new virtual env for this with python 3.7.4, still the same output for running examples folder. 

> python compare_license_texts.py
There was some problem with the function.
Traceback (most recent call last):
  File "compare_license_texts.py", line 15, in <module>
    stringtemplate = object_a.returnfinalstring()
  File "/Users/rlodha/spdx-python-licensematching/normalize_license_text/normalize_class.py", line 30, in returnfinalstring
    normalized_string = self.copyrightsymbol(normalized_string)
  File "/Users/rlodha/spdx-python-licensematching/normalize_license_text/normalize_class.py", line 89, in copyrightsymbol
    normalized_string = normalized_string.replace("copyright", '')
AttributeError: 'NoneType' object has no attribute 'replace'